### PR TITLE
Rosdep import

### PR DESCRIPTION
**Pull request by planthaber**
Auto-imports rosdep definitions based on selected ros vserion

### CODE REVIEW

**Comment by chhtz** [comment]
Under Ubuntu 24.04 with a new bootstrap and just adding this package set:

```
  - github: dfki-ric/ros2-package_set
    branch: rosdep_import
```

I'm currently getting:

```
  git:https://github.com/dfki-ric/ros2-package_set.git branch=rosdep_import interactive=false push_to=git@github.com:/dfki-ric/ros2-package_set.git repository_id=github:/dfki-ric/ros2-package_set.git retry_count=10: checking out branch rosdep_import
  checked out git:https://github.com/dfki-ric/ros2-package_set.git branch=rosdep_import interactive=false push_to=git@github.com:/dfki-ric/ros2-package_set.git repository_id=github:/dfki-ric/ros2-package_set.git retry_count=10
  Which ros version should be used to import paxckage depenencies [humble, jazzy, rolling] ? [humble] jazzy
Importing rosdep to /home/dfki.uni-bremen.de/chertzberg/workspace/ros_test/autoproj/remotes/ros2/ubuntu.osdeps and /home/dfki.uni-bremen.de/chertzberg/workspace/ros_test/autoproj/remotes/ros2/ros.osdeps
  ERROR: init.rb(package_set=ros2):32: Alias parsing was not enabled. To enable it, pass `aliases: true` to `Psych::load` or `Psych::safe_load`.
```

**Comment by planthaber** [comment]
Refactored a bit and hopefully fixed the alias error (not present on 22.04)

**Comment by chhtz** [comment]
Thanks, it appears to work now (I'm getting an unrelated error that it (obviously) can't find `dfkigit: dfki-planning/ugv_nav4d` -- I think we wanted to change that url to the public github repository anyways ...)

**Comment by chhtz** [comment]
After your latest commits, this works out of the box on my Ubuntu24 now (including pulling osdeps):

```
package_sets:
  - github: dfki-ric/ros2-package_set
    branch: rosdep_import

layout:
  - slam/slam3d_ros2
  - planning/ugv_nav4d_ros2
```

**Comment by chhtz** [comment]
The only minor improvement I would see is automatically choosing as default the recommended ros-distro depending on the ubuntu version (22.04 -> humble, 24.04 -> jazzy).

File: init.rb [hunk 1]
```diff
@@ -4,10 +4,18 @@
 require_relative "./lib/rosdep_import.rb"
 
 # Declare optione to ask for the desired ros version
-# options must have an implementation in the import function of ./lib/rosdep_import.rb 
+# options must have an implementation in the import function of ./lib/rosdep_import.rb
+# set default based on OS
+os_names, os_versions = Autoproj.workspace.operating_system
+if os_names.include?('ubuntu')
+    if os_versions.include?('22.04')
+        Autoproj.config.set("ROS_VERSION", "humple")
```
Comment (chhtz) [inline init.rb]: A humble typo ;)

**Pull request merged**